hy/macros.py

```diff
@@ -123,13 +129,42 @@ def _import(module, module_name=module_name):
         _import(module)
 
 
-def make_emtpy_fn_copy(fn):
-    argspec = getargspec(fn)
-    formatted_args = formatargspec(*argspec)
-    fn_str = 'lambda {}: None'.format(
-        formatted_args.lstrip('(').rstrip(')'))
-
-    empty_fn = eval(fn_str)
+def make_empty_fn_copy(fn):
+    try:
+        argspec = getargspec(fn)
+    except:
+        return lambda *_: None
```

Comment:
@gilch `getargspec` won't work on functions Python thinks are builtins. Ultimately, the rest of this is needed anyway because of non-alphanumeric argument names (e.g. `(defn f [@@])`).

However, I probably should look into what the issue is regarding the `getargspec` call itself.